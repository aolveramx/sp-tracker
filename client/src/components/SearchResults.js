import { AlbumTrackImage } from './styles/SearchResults/AlbumTrackImage.styled'
import { TrackList } from './styles/SearchResults/TrackList.styled'
import { TrackListContainer } from './styles/SearchResults/TrackListContainer.styled'
import { TextMuted } from './styles/layout/TextMuted.styled'

const SearchResults = ({ track, chooseTrack }) => {
  const handleTrack = () => {
    chooseTrack(track)
  }

  return (
    <TrackList onClick={handleTrack}>
      <AlbumTrackImage src={track.albumUrl} />
      <TrackListContainer>
        <div>{track.title}</div>
        <TextMuted>{track.artist}</TextMuted>
      </TrackListContainer>
    </TrackList>
  )
}

export default SearchResults
