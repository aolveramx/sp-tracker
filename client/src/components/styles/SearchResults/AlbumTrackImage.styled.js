import styled from 'styled-components'

export const AlbumTrackImage = styled.img`
  height: 64px;
  width: 64px;
`