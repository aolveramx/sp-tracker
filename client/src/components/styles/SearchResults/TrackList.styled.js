import styled from 'styled-components'

export const TrackList = styled.div`
  cursor: pointer;
  display: flex;
  margin: 2rem;
  align-items: center;
`