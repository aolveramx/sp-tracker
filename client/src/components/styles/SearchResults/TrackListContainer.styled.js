import styled from 'styled-components'

export const TrackListContainer = styled.div`
  margin-left: 3rem;
`