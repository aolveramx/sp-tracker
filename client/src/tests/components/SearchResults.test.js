import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import SearchResults from '../../components/SearchResults';


describe('Tests in SearchResults.js', () => {

  test('should render <SearchResults />', () => {

    const track = {
      albumUrl: 'https://i.scdn.co/image/ab67616d00004851f9171c2b7bab0956cdfbd1fa',
      title: 'New Lands',
      artist: 'Justice',
    }

    const chooseTrack = jest.fn()

    const wrapper = shallow(<SearchResults track={track} chooseTrack={chooseTrack} />)
    expect(wrapper).toMatchSnapshot()
    
  });
  
});