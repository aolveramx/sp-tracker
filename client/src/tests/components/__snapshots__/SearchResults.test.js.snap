// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in SearchResults.js should render <SearchResults /> 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.img
    src="https://i.scdn.co/image/ab67616d00004851f9171c2b7bab0956cdfbd1fa"
  />
  <styled.div>
    <div>
      New Lands
    </div>
    <styled.div>
      Justice
    </styled.div>
  </styled.div>
</styled.div>
`;
